projectroles/management/commands/cleanappsettings.py

```diff
@@ -42,10 +42,29 @@ def handle(self, *args, **options):
             else:
                 def_kwargs['app_name'] = 'projectroles'
             try:
-                app_settings.get_definition(**def_kwargs)
+                definition = app_settings.get_definition(**def_kwargs)
+                if s.project and s.project.type not in definition.get(
+                    'project_types', ['PROJECT']
+                ):
+                    logger.info(
+                        'Deleting "{}" from project "{}" - project type "{}" does not match allowed types: {}'.format(
+                            get_setting_str(s),
+                            s.project.title,
+                            s.project.type,
+                            definition.get('project_types', ['PROJECT']),
+                        )
+                    )
+                    s.delete()
+                elif not definition:
+                    logger.info(
+                        'Deleting "{}" from project "{}" - definition not found'.format(
+                            get_setting_str(s), s.project.title
+                        )
+                    )
+                    s.delete()
             except ValueError:
                 logger.info(
-                    'Deleting "{}" from project "{}"'.format(
+                    'Deleting "{}" from project "{}" - definition not found'.format(
```

Comment:
Here we check for the definition not existing twice, I would like us to remove the repetition. Handle the possibly missing definition first and continue with other checks if it's found.

Also, I don't believe it's possible for `get_definition()` to return `None`? It at least should raise a `ValueError` if not successful. So I'm not sure if we ever hit the condition above.